Duplicati/Library/Utility/OverrideableStream.cs

```diff
@@ -33,7 +33,7 @@ namespace Duplicati.Library.Utility
     /// </summary>
     public class OverrideableStream : Stream
     {
-        protected System.IO.Stream m_basestream;
+        protected System.IO.Stream? m_basestream;

```

Comment:
This one can never be null and should be `readonly`, it is being used as an `m_isDisposed` flag.
This removes all the other null checks.

Reply:
Changing that stream to readonly means the assignment here causes an error
<img width="230" alt="image" src="https://github.com/duplicati/duplicati/assets/60369677/8f72559d-9b13-452a-a2af-fba61e450078">